pages/article/ArticleHandler.inc.php

```diff
@@ -224,8 +285,12 @@ function downloadSuppFile($args, $request) {
 	 */
 	function download($args, $request) {
 		$articleId = isset($args[0]) ? $args[0] : 0;
-		$galleyId = isset($args[1]) ? $args[1] : 0;
-		$fileId = isset($args[2]) ? (int) $args[2] : 0;
+		$submissionRevision = isset($args[1]) ? (int) $args[1] : 0;
```

Comment:
Maybe to provide the submissionRevision in a different way to the download function and optional? -- so that the function could be self contained i.e. that it could be called directly and not only from the function view. Then the redirect could stay and you could only provide the parameter if necessary?